packages/dds/tree/src/test/changeset/visit.spec.ts

```diff
@@ -0,0 +1,217 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { Value } from "../..";
+import { Delta, DeltaVisitor, visitDelta } from "../../changeset";
+import { deepFreeze } from "../utils";
+
+function visit(delta: Delta.Root, visitor: DeltaVisitor): void {
+    deepFreeze(delta);
+    visitDelta(delta, visitor);
+}
+
+type VisitScript = VisitCall[];
+
+type VisitCall =
```

Comment:
I think it is possible to generate this from the Visitor type using keyof ( https://www.typescriptlang.org/docs/handbook/2/keyof-types.html ) and indexed types ( https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html ) combined with Parameters ( https://www.typescriptlang.org/docs/handbook/utility-types.html#parameterstype ).

I'm not sure thats worth doing just for this, but I think you can make a generic type that takes an interface and produces a type that is a intersection of all calls you could make to it this way. Might be a nice general tool to have around (or might already exist).